src/analyses/base.ml

```diff
@@ -2065,17 +2065,23 @@ struct
       let is_cmp = function
         | BinOp ((Lt | Gt | Le | Ge | Eq | Ne), _, _, t) -> true
         | _ -> false
-    in
-        try
-          let ik = Cilfacade.get_ikind_exp exp in
-          let itv = if not tv || is_cmp exp then (* false is 0, but true can be anything that is not 0, except for comparisons which yield 1 *)
+      in
+      try
+        let ik = Cilfacade.get_ikind_exp exp in
+        let itv = if not tv || is_cmp exp then (* false is 0, but true can be anything that is not 0, except for comparisons which yield 1 *)
             ID.of_bool ik tv (* this will give 1 for true which is only ok for comparisons *)
           else
             ID.of_excl_list ik [BI.zero] (* Lvals, Casts, arithmetic operations etc. should work with true = non_zero *)
-          in
-          inv_exp (`Int itv) exp st
-        with Invalid_argument _ ->
-          inv_exp (`Float (FD.top_of (Cilfacade.get_fkind_exp exp))) exp st
+        in
+        inv_exp (`Int itv) exp st
+      with Invalid_argument _ ->
+        let fk = Cilfacade.get_fkind_exp exp in
```

Comment:
In C, the expressions in a condition may also have pointer type, i.e. neither be an integer nor a float; so this may throw an exception.